nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -595,29 +595,58 @@ public static void checkTypeParameterNullnessForConditionalExpression(
    * Checks that for each parameter p at a call, the type parameter nullability for p's type matches
    * that of the corresponding formal parameter. If a mismatch is found, report an error.
    *
-   * @param formalParams the formal parameters
-   * @param actualParams the actual parameters
+   * @param methodSymbol the symbol for the method being called
+   * @param tree the tree representing the method call
+   * @param actualParams the actual parameters at the call
    * @param isVarArgs true if the call is to a varargs method
    * @param analysis the analysis object
    * @param state the visitor state
    */
   public static void compareGenericTypeParameterNullabilityForCall(
-      List<Symbol.VarSymbol> formalParams,
+      Symbol.MethodSymbol methodSymbol,
+      Tree tree,
       List<? extends ExpressionTree> actualParams,
       boolean isVarArgs,
       NullAway analysis,
       VisitorState state) {
     if (!analysis.getConfig().isJSpecifyMode()) {
       return;
     }
-    int n = formalParams.size();
+    Type invokedMethodType = methodSymbol.type;
+    if (!methodSymbol.isStatic() && tree instanceof MethodInvocationTree) {
+      ExpressionTree methodSelect = ((MethodInvocationTree) tree).getMethodSelect();
+      Type enclosingType;
+      if (methodSelect instanceof MemberSelectTree) {
+        enclosingType = getTreeType(((MemberSelectTree) methodSelect).getExpression(), state);
+      } else {
+        // implicit this parameter
+        enclosingType = methodSymbol.owner.type;
+      }
+      if (enclosingType != null) {
+        invokedMethodType = state.getTypes().memberType(enclosingType, methodSymbol);
+      }
+    }
+    // Handle generic methods
+    if (tree instanceof MethodInvocationTree && methodSymbol.type instanceof Type.ForAll) {
+      MethodInvocationTree methodInvocationTree = (MethodInvocationTree) tree;
+
+      List<? extends Tree> typeArgumentTrees = methodInvocationTree.getTypeArguments();
+      com.sun.tools.javac.util.List<Type> explicitTypeArgs = convertTreesToTypes(typeArgumentTrees);
+
+      Type.ForAll forAllType = (Type.ForAll) methodSymbol.type;
+      Type.MethodType underlyingMethodType = (Type.MethodType) forAllType.qtype;
+      invokedMethodType =
+          state.getTypes().subst(underlyingMethodType, forAllType.tvars, explicitTypeArgs);
+    }
```

Comment:
This exact logic (or very similar) now appears in multiple places.  Can we extract it into a method that returns a method type with the generic type arguments substituted in?